File: sdks/sdkwork-craw-chat-sdk-admin/sdkwork-craw-chat-sdk-admin-typescript/composed/src/social-runtime-module.ts
import type { JsonObject } from './types.js';
import type { CrawChatAdminSdkContext } from './sdk-context.js';

export class CrawChatAdminSocialRuntimeModule {
  constructor(private readonly context: CrawChatAdminSdkContext) {}

  claimPendingTargeted(body: JsonObject): Promise<JsonObject> {
    return this.context.backendClient.socialRuntime.claimPendingSharedChannelSyncTargeted(body);
  }

  getDeadLetterInventory(): Promise<JsonObject> {
    return this.context.backendClient.socialRuntime.getDeadLetterSharedChannelSyncInventory();
  }

  getDeliveredInventory(): Promise<JsonObject> {
    return this.context.backendClient.socialRuntime.getDeliveredSharedChannelSyncInventory();
  }

  getDeliveryStateInventory(): Promise<JsonObject> {
    return this.context.backendClient.socialRuntime.getSharedChannelSyncDeliveryStateInventory();
  }

  getPendingInventory(): Promise<JsonObject> {
    return this.context.backendClient.socialRuntime.getPendingSharedChannelSyncInventory();
  }

  reclaimStalePending(): Promise<JsonObject> {
    return this.context.backendClient.socialRuntime.reclaimStalePendingSharedChannelSync();
  }

  releasePendingTargeted(body: JsonObject): Promise<JsonObject> {
    return this.context.backendClient.socialRuntime.releasePendingSharedChannelSyncTargeted(body);
  }

  repairSnapshot(): Promise<JsonObject> {
    return this.context.backendClient.socialRuntime.repairSocialRuntimeSnapshot();
  }

  repairSharedChannelSync(): Promise<JsonObject> {
    return this.context.backendClient.socialRuntime.repairSharedChannelSync();
  }

  republishPendingTargeted(body: JsonObject): Promise<JsonObject> {
    return this.context.backendClient.socialRuntime.republishPendingSharedChannelSyncTargeted(body);
  }

  requeueDeadLetters(): Promise<JsonObject> {
    return this.context.backendClient.socialRuntime.requeueDeadLetterSharedChannelSync();
  }

  requeueDeadLettersTargeted(body: JsonObject): Promise<JsonObject> {
    return this.context.backendClient.socialRuntime.requeueDeadLetterSharedChannelSyncTargeted(body);
  }

  takeoverPendingTargeted(body: JsonObject): Promise<JsonObject> {
    return this.context.backendClient.socialRuntime.takeoverPendingSharedChannelSyncTargeted(body);
  }
}

File: sdks/sdkwork-craw-chat-sdk-admin/sdkwork-craw-chat-sdk-admin-flutter/composed/lib/src/social_runtime_module.dart
import 'package:craw_chat_admin_backend_sdk/craw_chat_admin_backend_sdk.dart';

import 'context.dart';

class CrawChatAdminSocialRuntimeModule {
  final CrawChatAdminSdkContext context;

  CrawChatAdminSocialRuntimeModule(this.context);

  Future<JsonObject> claimPendingTargeted(JsonObject body) {
    return context.backendClient.socialRuntime
        .claimPendingSharedChannelSyncTargeted(body);
  }

  Future<JsonObject> getDeadLetterInventory() {
    return context.backendClient.socialRuntime
        .getDeadLetterSharedChannelSyncInventory();
  }

  Future<JsonObject> getDeliveredInventory() {
    return context.backendClient.socialRuntime
        .getDeliveredSharedChannelSyncInventory();
  }

  Future<JsonObject> getDeliveryStateInventory() {
    return context.backendClient.socialRuntime
        .getSharedChannelSyncDeliveryStateInventory();
  }

  Future<JsonObject> getPendingInventory() {
    return context.backendClient.socialRuntime
        .getPendingSharedChannelSyncInventory();
  }

  Future<JsonObject> reclaimStalePending() {
    return context.backendClient.socialRuntime
        .reclaimStalePendingSharedChannelSync();
  }

  Future<JsonObject> releasePendingTargeted(JsonObject body) {
    return context.backendClient.socialRuntime
        .releasePendingSharedChannelSyncTargeted(body);
  }

  Future<JsonObject> repairSnapshot() {
    return context.backendClient.socialRuntime.repairSocialRuntimeSnapshot();
  }

  Future<JsonObject> repairSharedChannelSync() {
    return context.backendClient.socialRuntime.repairSharedChannelSync();
  }

  Future<JsonObject> republishPendingTargeted(JsonObject body) {
    return context.backendClient.socialRuntime
        .republishPendingSharedChannelSyncTargeted(body);
  }

  Future<JsonObject> requeueDeadLetters() {
    return context.backendClient.socialRuntime
        .requeueDeadLetterSharedChannelSync();
  }

  Future<JsonObject> requeueDeadLettersTargeted(JsonObject body) {
    return context.backendClient.socialRuntime
        .requeueDeadLetterSharedChannelSyncTargeted(body);
  }

  Future<JsonObject> takeoverPendingTargeted(JsonObject body) {
    return context.backendClient.socialRuntime
        .takeoverPendingSharedChannelSyncTargeted(body);
  }
}
